apps/networkmonitor/networkmonitor.nim

```diff
@@ -82,43 +99,68 @@ proc setConnectedPeersMetrics(discoveredNodes: seq[Node],
     let ip = $typedRecord.get().ip.get().join(".")
     allPeers[peerId].ip = ip
 
-    let peer = toRemotePeerInfo(discNode.record)
-    if not peer.isOk():
-      warn "error converting record to remote peer info", record=discNode.record
-      continue
+    # try to ping the peer
+    if getTime().toUnix() >= allPeers[peerId].lastTimeConnected + ReconnectTime and allPeers[peerId].retries < MaxConnectionRetries:
```

Comment:
I'd suggest moving the logic under this out into a separate function. `setConnectedPeersMetrics` is becoming diffult to follow and maintain.
Something like:
```
if dueForPing(allPeers[PeerId]):
  let pingDelay = ping(allPeers[PeerId]).onError ...
```

My not be as simple as that, but I guess we should start separating concerns here as much as possible.